packages/kbn-health-gateway-server/src/kibana/kibana_config.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { readFileSync } from 'fs';
+import type { Duration } from 'moment';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
+
+const hostURISchema = schema.uri({ scheme: ['http', 'https'] });
+
+const configSchema = schema.object({
+  hosts: schema.arrayOf(hostURISchema, {
+    minSize: 1,
+  }),
+  requestTimeout: schema.duration({ defaultValue: '30s' }),
+  ssl: schema.object({
+    verificationMode: schema.oneOf(
+      [schema.literal('none'), schema.literal('certificate'), schema.literal('full')],
+      { defaultValue: 'full' }
+    ),
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.string(), schema.arrayOf(schema.string(), { minSize: 1 })])
+    ),
+    certificate: schema.maybe(schema.string()),
+  }),
+});
+
+export type KibanaConfigType = TypeOf<typeof configSchema>;
+
+export const config: ServiceConfigDescriptor<KibanaConfigType> = {
+  path: 'kibana' as const,
+  schema: configSchema,
+};
+
+export class KibanaConfig {
+  /**
+   * Kibana hosts that the gateway will connect to.
+   */
+  public readonly hosts: string[];
+
+  /**
+   * Timeout after which HTTP requests to the Kibana hosts will be aborted.
+   */
+  public readonly requestTimeout: Duration;
+
+  /**
+   * Settings to configure SSL connection between the gateway and Kibana hosts.
+   */
+  public readonly ssl: SslConfig;
+
+  constructor(rawConfig: KibanaConfigType) {
+    this.hosts = rawConfig.hosts;
+    this.requestTimeout = rawConfig.requestTimeout;
+
+    const { verificationMode } = rawConfig.ssl;
```

Comment:
I think this should be `rawConfig.ssl.verificationMode`

Reply:
This is most likely causing the issues we are seeing during testing, the default value is `full` https://github.com/elastic/kibana/pull/141172/files#diff-be5973840f6b4995df276db5c68948dcd599d8897988adf6826e60830e0d3f6bR24
so it's ignoring `certificate` and `certificateAuthorities`

Reply:
The only reason we use a default value of `full` here is for consistency with the ssl config settings Kibana uses for connecting to ES: https://github.com/elastic/kibana/blob/a159263ceff9f4e19c307334d0824a8fb33a01f8/packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_config.ts#L111-L114

However, I believe in the cloud stackpacks we are explicitly setting this to `certificate`, correct? (or at least I think we did when @kobelb merged the cloud assets changes)

When running from source locally, I'm able to use TLS with Kibana & ES and still get the expected response from the health gateway with these config fields:
```yaml
server:
  port: 3000
  host: 'localhost'
  ssl:
    enabled: true
    key: /usr/share/kibana/config/certs/kibana.key
    certificate: /usr/share/kibana/config/certs/kibana.crt
    certificateAuthorities: /usr/share/kibana/config/certs/ca.crt
kibana:
  hosts:
    - 'https://localhost:5605'
    - 'https://localhost:5606'
  ssl:
    certificateAuthorities: /usr/share/kibana/config/certs/ca.crt
    verificationMode: certificate
logging:
  root:
    appenders: ['console']
    level: 'all'
```

(cc @dokmic)